Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -419,18 +417,15 @@ namespace AzToolsFramework::ComponentModeFramework
     void ComponentModeSwitcher::AfterUndoRedo()
     {
         // wait one frame for the undo stack to actually be updated
-        QTimer::singleShot(
-            0,
-            nullptr,
+        AZ::TickBus::QueueFunction(
             [this]()
             {
-                auto* toolsApplicationRequests = AzToolsFramework::ToolsApplicationRequestBus::FindFirstHandler();
-                if (toolsApplicationRequests != nullptr)
+                if (auto* toolsApplicationRequests = AzToolsFramework::ToolsApplicationRequestBus::FindFirstHandler())
                 {
                     const auto& selectedEntityIds = toolsApplicationRequests->GetSelectedEntities();
                     if (selectedEntityIds.size() == 1)
```

Comment:
what is this check for, I guess might want to add a note maybe?

Reply:
I'll check with @amzn-ahmadkrm about this :+1: